### PR TITLE
[FIX] User can't invite or join other Omnichannel rooms

### PR DESCRIPTION
**Pull request by renatobecker**
## Proposed changes
Omnichannel users(admins) weren't able to invite other agents to join the chat. This is a very useful feature, so community, and also customers were complaining about the issue.

## Issue(s)
Closes https://github.com/RocketChat/Rocket.Chat/issues/18023

## How to test or reproduce
- Access an Omnichannel Chat
- Use the `/invite` command to invite other agents to join the chat
- Nothing happens

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Hotfix (a major bugfix that has to be merged asap)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/livechat/lib/LivechatRoomType.js [hunk 1]
```diff
@@ -85,6 +85,10 @@ export default class LivechatRoomType extends RoomTypeConfig {
 		}
 	}
 
+	allowMemberAction(room, action) {
+		return action === RoomMemberActions.INVITE || RoomMemberActions.JOIN;
```
Comment (sampaiodiego) [inline app/livechat/lib/LivechatRoomType.js:89]: I guess you want to test if `action` is either `INVITE` OR `JOIN` right? in this case, this code doesn't do that.. what you could do is something like:
```suggestion
		return [RoomMemberActions.INVITE, RoomMemberActions.JOIN].includes(action);
```

File: app/lib/server/functions/addUserToRoom.js [hunk 2]
```diff
@@ -21,6 +22,10 @@ export const addUserToRoom = function(rid, user, inviter, silenced) {
 		return;
 	}
 
+	if (room.t === 'l' && !hasPermission(user._id, 'view-l-room')) {
```
Comment (sampaiodiego) [inline app/lib/server/functions/addUserToRoom.js:25]: please move this code to `allowMemberAction` method of the livechat room type.. the idea behind it is specifically to not have this kind specific code spread across common function.

**Review by sampaiodiego — CHANGES_REQUESTED** [review]

File: app/lib/server/functions/addUserToRoom.js (hunk 2 above)
Comment (renatobecker) [reply app/lib/server/functions/addUserToRoom.js:25]: Yeah, that was the idea but the signature of the `allowMemberAction` method doesn't support the user who is going to join the room, which can be added, then I can move the code to `allowMemberAction` method and check all conditions there.

Comment (sampaiodiego) [reply app/lib/server/functions/addUserToRoom.js:25]: no, we didn't 😬  we have a `beforeLeaveRoom` https://github.com/RocketChat/Rocket.Chat/blob/b469c18d39bf867af4da1e75e55bd6bee6e15fd5/app/livechat/server/startup.js#L77

Comment (renatobecker) [reply app/lib/server/functions/addUserToRoom.js:25]: Oh nice! The callback seems to be a good option here. 
Let me change the code and see how it works! :D

Comment (renatobecker) [reply app/lib/server/functions/addUserToRoom.js]: @sampaiodiego the fixed based on callback is done!

**Review by sampaiodiego — APPROVED** [review]

**Pull request merged**